subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -190,15 +199,141 @@ int bt_cap_initiator_broadcast_get_base(struct bt_cap_broadcast_source *source,
 #endif /* CONFIG_BT_BAP_BROADCAST_SOURCE */
 
 #if defined(CONFIG_BT_BAP_UNICAST_CLIENT)
-static const struct bt_uuid *cas_uuid = BT_UUID_CAS;
+enum {
+	CAP_UNICAST_PROC_STATE_ACTIVE,
+	CAP_UNICAST_PROC_STATE_ABORTED,
+
+	CAP_UNICAST_PROC_STATE_FLAG_NUM,
+} cap_unicast_proc_state;
+
+enum cap_unicast_subproc_type {
+	CAP_UNICAST_SUBPROC_TYPE_NONE,
+	CAP_UNICAST_SUBPROC_TYPE_CODEC_CONFIG,
+	CAP_UNICAST_SUBPROC_TYPE_QOS_CONFIG,
+	CAP_UNICAST_SUBPROC_TYPE_ENABLE,
+	CAP_UNICAST_SUBPROC_TYPE_START,
+	CAP_UNICAST_SUBPROC_TYPE_META_UPDATE,
+	CAP_UNICAST_SUBPROC_TYPE_RELEASE,
+};
+
+struct cap_unicast_proc {
+	ATOMIC_DEFINE(proc_state_flags, CAP_UNICAST_PROC_STATE_FLAG_NUM);
+	/* Total number of streams in the procedure*/
+	size_t stream_cnt;
+	/* Number of streams where a subprocedure have been started */
+	size_t stream_initiated;
+	/* Number of streams done with the procedure */
+	size_t stream_done;
```

Comment:
Would it make sense to postfix `stream_initiated` and `stream_done` with `_cnt` to more clearly indicate that they are counters and will be compared with `stream_cnt` (~ logically connected)